packages/server/tests/acceptance/hbarLimiter.spec.ts

```diff
@@ -706,6 +745,125 @@ describe('@hbarlimiter HBAR Limiter Acceptance Tests', function () {
             reusableTestsForNonBasicTiers(SubscriptionTier.PRIVILEGED, maxPrivilegedSpendingLimit);
           });
         });
+
+        describe('@hbarlimiter-batch2 Multiple users with different tiers', () => {
+          interface AliasAccountPlan {
+            aliasAccounts: AliasAccount[];
+            hbarSpendingPlan: IDetailedHbarSpendingPlan;
+          }
+
+          let accountPlanObject: Record<SubscriptionTier, AliasAccountPlan[]>;
+
+          const accountPlanRequirements: Record<SubscriptionTier, number> = {
+            BASIC: 2,
+            EXTENDED: 2,
+            PRIVILEGED: 2,
+          };
+
+          const createMultipleAliasAccountsWithSpendingPlans = async (
+            accountPlanRequirements: Record<SubscriptionTier, number>,
+          ) => {
+            const accountPlanObject: Record<SubscriptionTier, AliasAccountPlan[]> = {
+              BASIC: [],
+              EXTENDED: [],
+              PRIVILEGED: [],
+            };
+
+            for (const [subscriptionTier, numOfAccounts] of Object.entries(accountPlanRequirements)) {
+              for (let i = 0; i < numOfAccounts; i++) {
+                const accountCreatedResult = await createAliasAndAssociateSpendingPlan(
+                  subscriptionTier as SubscriptionTier,
+                );
+                accountPlanObject[subscriptionTier].push(accountCreatedResult);
+              }
+            }
+
+            return accountPlanObject;
+          };
+
+          before(async () => {
+            accountPlanObject = await createMultipleAliasAccountsWithSpendingPlans(accountPlanRequirements);
+          });
+
+          it('should individually update amountSpents of different spending plans', async () => {
+            const callingAccountAddresses: string[] = [];
+            const allAccountAliases = Object.values(accountPlanObject)
+              .flat()
+              .map((accountAliasPlan) => accountAliasPlan.aliasAccounts[0]);
+
+            const deployPromises = allAccountAliases.map(async (accountAlias) => {
+              if (Math.random() < 0.5) {
+                const tx = await deployContract(mediumSizeContract, accountAlias.wallet);
+                await tx.waitForDeployment();
+                return accountAlias.address; // Return the address for those that made deployments
+              }
+              return null;
+            });
+
+            const results = await Promise.all(deployPromises);
+            callingAccountAddresses.push(...results.filter((address) => address !== null));
+
+            expect(callingAccountAddresses.length).to.gt(0);
+
+            // awaiting for HBAR limiter to finish updating expenses in the background
+            await Utils.wait(6000);
```

Comment:
Can the expenses be queried here instead of just waiting?

Reply:
Nice point updated with polling